src/drivers/clock/epit/epit.c

```diff
@@ -0,0 +1,99 @@
+#define EPIT_TARGET_HZ  1000
+
+#define EPIT_CR 		(EPIT_BASE + 0x00) /* Control Register */
+#define EPIT_SR     	(EPIT_BASE + 0x04) /* Status Register */
+#define EPIT_LR     	(EPIT_BASE + 0x08) /* Load Register */
+#define EPIT_CMPR   	(EPIT_BASE + 0x0C) /* Compare Register */
+#define EPIT_CNR    	(EPIT_BASE + 0x10) /* Counter Register */
+
+#define EPIT_CR_PRESCLR (0x42 << 4)/* Prescaler value of 66 */
+#define EPIT_CR_EN 		(1 << 0) /* Enable Bit */
+#define EPIT_CR_ENMOD   (1 << 1) /* Enable Mode Bit */
+#define EPIT_CR_OCIEN   (1 << 2) /* Output compare Interrupt Enable bit*/
+#define EPIT_CR_RLD		(1 << 3) /* Counter Reload Control bit */
+#define EPIT_SR_CLEAR	(1 << 0) /* Clear the status register*/
+#define EPIT_CR_CLKSRC	(1 << 24) /* Use peripheral clock as input, 66MHz */
+#define EPIT_CR_IOVW	(1 << 17) /* Load Immidiate Override*/
+
+static struct clock_source epit_clock_source;
+static irq_return_t clock_handler(unsigned int irq_nr, void *data) {
+	clock_tick_handler(irq_nr, data);
+	REG32_STORE(EPIT_SR, EPIT_SR_CLEAR);
+	return IRQ_HANDLED;
+}
+
+static int epit_init(void) {
+	clock_source_register(&epit_clock_source);
+
+	return irq_attach(EPIT_IRQ,
+	                  clock_handler,
+	                  0,
+	                  &epit_clock_source,
+	                  "EPIT");
+}
+
+static int epit_config(struct time_dev_conf * conf) {
+	/* changin clock source to peripheral clock */
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) & (~EPIT_CR_OCIEN)); /* Turn off the interrupt */
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) & (~EPIT_CR_EN)); /* disable the timer*/
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) | EPIT_CR_CLKSRC); /* Select the clock source */
+	REG32_STORE(EPIT_SR, EPIT_SR_CLEAR); /* Clear the status register */
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) | EPIT_CR_ENMOD); /* Set enable mode = 1 */
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) | EPIT_CR_EN); /* enable the timer */  
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) | EPIT_CR_OCIEN); /* Enable Interrupts */
+
+	/* setting the load and compare registers */
+	REG32_STORE(EPIT_CR, REG32_LOAD(EPIT_CR) | EPIT_CR_RLD | EPIT_CR_IOVW | EPIT_CR_PRESCLR); /* Set reload and overrite */
+	REG32_STORE(EPIT_LR, 0x3e8); /* Load regiter set to 1000 */
+	REG32_STORE(EPIT_CMPR, 0x0); /* Compare register set to 0 */
+	return 0;
+}
+
+static cycle_t epit_read(void) {
+	return REG32_LOAD(EPIT_CNR);
+}
+
+static struct time_event_device epit_event = {
+	.config   = epit_config,
+	.event_hz = EPIT_TARGET_HZ,
+	.irq_nr   = EPIT_IRQ,
+};
+
+static struct time_counter_device epit_counter = {
+	.read     = epit_read,
+	.cycle_hz = EPIT_TARGET_HZ,
+};
+
+static struct clock_source epit_clock_source = {
+	.name           = "epit",
+	.event_device   = &epit_event,
+	.counter_device = &epit_counter,
+	.read           = clock_source_read,
+};
+
+EMBOX_UNIT_INIT(epit_init);
+
+STATIC_IRQ_ATTACH(EPIT_IRQ, clock_handler, &imx6_clock_source);
+
+PERIPH_MEMORY_DEFINE(epit_mem, EPIT_BASE, 0x10);
```

Comment:
Are you sure `0x10`? How many registers EPIT has? I see you provide 5 registers, but even 5 regs have `5 * 4 = 20 = 0x14` in size.

Reply:
I am sorry for this, the EPIT does have 5 registers, I will change It to 0x14.